lambdatimes/src/components/Content/Content.js

```diff
@@ -1,56 +1,54 @@
 import React, { Component } from 'react';
-
 import Tabs from './Tabs';
 import Cards from './Cards';
-
-// Importing our tab and card data. No need to change anything here.
 import { tabData, cardData } from '../../data';
+import Styled from 'styled-components';
+
+const ContentContainer = Styled.div`
+  display: flex; 
+  flex-direction: column; 
+  align-items: center;
+`
 
 export default class Content extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      selected: 'all',
+      selectedTab: 'all',
       tabs: [],
       cards: []
     };
   }
 
   componentDidMount() {
-    // Once the component has mounted, get the data and reflect that data on the state.
+    this.setState({tabs:tabData,cards:cardData})
   }
 
-  changeSelected = tab => {
-    // this function should take in the tab and update the state with the new tab.
-  };
+  selectTabHandler = event =>{
+    this.setState({selectedTab:event.target.innerText})
+  }
 
   filterCards = () => {
-    /* Right now this function only returns the cards on state.
-      We're going to make this function more dynamic
-      by using it to filter out our cards for when a tab is selcted
-      
-      Notice that we're passing this function to our <Cards /> component below.
-      This function returns an array of cards, so we can just pass it down as such.
 
-      Your algorithim for the logic here is as follows: 
-        - if the selected tab is 'all' it should return all 
-          of the items from cardData. 
-        - else, it should only return those cards whose 'tab' matched this.state.selected.
-    */
-    return this.state.cards;
+    if(this.state.selectedTab.toLowerCase() ==='all'){return this.state.cards;}
+    return this.state.cards.filter(element =>{
+        if(element.tab.toLowerCase() === this.state.selectedTab.toLowerCase()){
```

Comment:
you don't actually need to return the element, just a truthy or falsy value is enough to tell it what to keep. So try this:

`return this.state.cards.filter(element => (element.tab.toLowerCase() === this.state.selectedTab.toLowerCase()))`